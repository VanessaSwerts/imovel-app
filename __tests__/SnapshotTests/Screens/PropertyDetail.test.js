import React from 'react'
import { act, create } from 'react-test-renderer'
import PropertyDetail from '../../../src/screens/PropertyDetail'

jest.useFakeTimers()
jest.mock('@expo/vector-icons/FontAwesome5', () => 'Icon')
jest.mock('@react-native-community/async-storage', () => 'AsyncStorage')
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn()
    }),
    useRoute: () => ({
      route: jest.fn()
    })
  }
})

describe('PropertyDetail test', () => {
  it('render PropertyDetail screen correctly', () => {
    const tree = create(<PropertyDetail />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
