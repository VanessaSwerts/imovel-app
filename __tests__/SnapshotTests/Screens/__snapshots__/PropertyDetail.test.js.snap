// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetail test render PropertyDetail screen correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#14213D",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    scrollsToTop={true}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          activeDot={
            <View
              style={
                Object {
                  "backgroundColor": "#FCA311",
                  "borderRadius": 5,
                  "height": 10,
                  "margin": 3,
                  "width": 10,
                }
              }
            />
          }
          automaticallyAdjustContentInsets={false}
          autoplay={false}
          autoplayDirection={true}
          autoplayTimeout={2.5}
          bounces={false}
          contentContainerStyle={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "height": 280,
              },
            ]
          }
          contentOffset={
            Object {
              "x": 0,
            }
          }
          disableNextButton={false}
          disablePrevButton={false}
          dot={
            <View
              style={
                Object {
                  "backgroundColor": "#b2b2b2",
                  "borderRadius": 5,
                  "height": 10,
                  "margin": 3,
                  "width": 10,
                }
              }
            />
          }
          horizontal={true}
          index={0}
          loadMinimal={false}
          loadMinimalSize={1}
          loop={true}
          onIndexChanged={[Function]}
          onMomentumScrollEnd={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          paginationStyle={
            Object {
              "bottom": null,
              "left": null,
              "right": 15,
              "top": 15,
            }
          }
          pagingEnabled={true}
          removeClippedSubviews={true}
          scrollsToTop={false}
          showsButtons={false}
          showsHorizontalScrollIndicator={false}
          showsPagination={true}
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "height": 1334,
                    "width": 750,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderTopLeftRadius": 50,
            "marginTop": -15,
            "minHeight": 450,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "#FCA311",
              "fontSize": 22,
              "fontWeight": "700",
              "paddingBottom": 5,
              "paddingTop": 10,
              "textAlign": "right",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 17,
              "fontWeight": "500",
              "lineHeight": 20,
              "textAlign": "right",
            }
          }
        >
          , 
        </Text>
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 17,
              "fontWeight": "500",
              "lineHeight": 20,
              "textAlign": "right",
            }
          }
        >
           - 
        </Text>
        <Text
          style={
            Object {
              "color": "#14213D",
              "fontSize": 18,
              "fontWeight": "600",
              "paddingBottom": 5,
              "paddingTop": 20,
            }
          }
        >
          Sobre esse imóvel:
        </Text>
        <Text
          numberOfLines={5}
          style={
            Object {
              "color": "#999",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 18,
              "marginBottom": 10,
              "textAlign": "justify",
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 20,
              "marginTop": 20,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#14213D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#14213D"
                name="bed"
                size={18}
              />
            </View>
            <Text>
               quartos
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#14213D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#14213D"
                name="bath"
                size={20}
              />
            </View>
            <Text>
               banheiro
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 20,
              "marginTop": 20,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#14213D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#14213D"
                name="ruler-horizontal"
                size={18}
              />
            </View>
            <Text>
               m²
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderColor": "#14213D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#14213D"
                name="dog"
                size={19}
              />
            </View>
            <Text>
              Não aceita pet
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 15,
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#14213D",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 5,
                  "paddingTop": 20,
                }
              }
            >
              Aluguel:
            </Text>
            <Text
              numberOfLines={5}
              style={
                Object {
                  "color": "#999",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "paddingLeft": 8,
                }
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "backgroundColor": "#FCA311",
                "borderRadius": 8,
                "height": 35,
                "justifyContent": "center",
                "opacity": 1,
                "width": 160,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#14213D",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Entrar em contato
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 15,
            "opacity": 1,
            "position": "absolute",
            "top": 15,
            "zIndex": 5,
          }
        }
      >
        <Icon
          color="#FCA311"
          name="arrow-left"
          size={20}
        />
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#e6e6e6",
                "borderColor": "#14213D",
                "borderRadius": 24,
                "borderWidth": 0.5,
                "elevation": 5,
                "padding": 18,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Object {
                    "backgroundColor": "#14213D",
                    "borderColor": "#14213D",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "height": 80,
                    "width": 80,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "justifyContent": "center",
                    "paddingLeft": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#14213D",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    Proprietário:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    Bianca Souza
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#14213D",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    Telefone:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    (35) 98877-6655
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#14213D",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    E-mail:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    biancasouza@gmail.com
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-end",
                  "backgroundColor": "#FCA311",
                  "borderRadius": 8,
                  "height": 30,
                  "justifyContent": "center",
                  "width": 80,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#14213D",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Fechar
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
